pkg/compiler/analysis.go

```diff
@@ -340,6 +352,89 @@ func (c *codegen) analyzeFuncUsage() funcUsage {
 				}
 				nodeCache[name] = declPair{n, c.importMap, pkgPath}
 				return false // will be processed in the next stage
+			case *ast.GenDecl:
+				// After skipping all funcDecls, we are sure that each value spec
+				// is a globally declared variable or constant. We need to gather global
+				// vars from both main and imported packages.
+				if n.Tok == token.VAR {
+					for i, s := range n.Specs {
+						valSpec := s.(*ast.ValueSpec)
+						for j, id := range valSpec.Names {
+							if id.Name != "_" {
+								name := c.getIdentName(pkgPath, id.Name)
+								globalVarsCache[name] = globalVar{
+									decl:      n,
+									specIdx:   i,
+									varIdx:    j,
+									ident:     id,
+									importMap: c.importMap,
+									path:      pkgPath,
+								}
+							}
+							// Traverse both named/unnamed global variables, check whether function call is present inside variable
+							// value and if so, mark all its children for further traversal. Example:
+							//  var A = 1
+							//  var B = A + f()		// B is unused but contains a function call, thus all B's children (which is A) will be marked "for further traversal".
+							//  func Main() int {
+							//  	return 1
+							//  }
+							//  func f() int {
+							//  	return 2
+							//  }`
+							if len(valSpec.Values) == 0 {
+								continue
+							}
+							var (
+								hasFuncCall bool
+								children    = funcUsage{}
+							)
+							multiRet := len(valSpec.Values) != len(valSpec.Names) // var A, B = f() where func f() (int, int)
+							if j == 0 || !multiRet {
+								ast.Inspect(valSpec.Values[j], func(node ast.Node) bool {
```

Comment:
Ugh, I'm just stuck with it. @fyrchik, @roman-khimov, could you, please, see the `TestUnusedGlobalEdgeCase`? It contains an example of global var declarations where mixed expression is used as a value (function call + a set of other complicated calls). The idea is the following: I have a map of all global variables (used and unused). In this `ast.Inspect` statement I'd like to pick the whole set of global variables from expressions that contains function calls. E.g. from
```
var A = 5
var _ = A + f()
```
I need to pick `A` global variable.

The problem is that there are a lot of cases where global variable can appear, see the `TestUnusedGlobalEdgeCase`.